gspread/spreadsheet.py

```diff
@@ -739,3 +739,33 @@ def list_protected_ranges(self, sheetid):
             raise WorksheetNotFound("worksheet id {} not found".format(sheetid))
 
         return sheet.get("protectedRanges", [])
+
+    def get_all_worksheet_values(self, skip_worksheet_titles: list = None):
+        """Grabs all the data from all the worksheets in one API call. Skips any worksheets that were named in the
+        skip_worksheet_title param.
+
+        :param list skip_worksheet_titles: A list of worksheet titles to skip.
+        :returns: a dict of worksheet data with worksheet title as key
+        :rtype: dict
+        """
+
+        if skip_worksheet_titles is None:
+            skip_worksheet_titles = []
+
+        ranges = []
+
+        for worksheet in self.worksheets().worksheets():
+            if worksheet.title not in skip_worksheet_titles:
+                ranges.append(worksheet.title)
```

Comment:
Here we should protect the worksheet title using the util function `absolute_range_name`